runtime/src/drivers.rs

```diff
@@ -114,31 +105,110 @@ impl Drivers {
                 .map_err(|_| CaliptraError::RUNTIME_INITIALIZE_DPE_FAILED)?;
             let env = DpeEnv::<CptraDpeTypes> {
                 crypto,
-                platform: DpePlatform::new(locality, hashed_rt_pub_key, &mut cert_chain),
+                platform: DpePlatform::new(locality, hashed_rt_pub_key, &mut drivers.cert_chain),
             };
-            let dpe = Self::initialize_dpe(env, &mut pcr_bank, locality)?;
-            persistent_data.get_mut().dpe = dpe;
+            let dpe = Self::initialize_dpe(env, &mut drivers.pcr_bank, locality)?;
+            drivers.persistent_data.get_mut().dpe = dpe;
         }
 
+        if reset_reason != ResetReason::ColdReset {
+            let dpe = &drivers.persistent_data.get().dpe;
+            // Find root node by finding the context with parent equal to ROOT_INDEX
+            let root_idx = dpe
```

Comment:
On UpdateReset I think we also need to update the TCI at the root to match the new PCR3 value.

Reply:
Done